### DISCUSSION TITLE
Regarding `arg_lock` in v11.0.0

### DISCUSSION THREAD

**Discussion by garysassano** [devsectop/tf-via-pr#260] [General]
What if you just want lock false in the plan but lock true in the apply? Which is quite common, since you want multiple people to be able to run workflows at the same time to see the result of their changes, but you also need to lock the state when someone is modifying the configuration.

**Comment by rdhar** [comment]
@garysassano Good point, well-made, and easily addressed with a couple more examples as in #261, eh?

**Reply by garysassano** [reply]
That works!

Alternatively, you could have modified the `arg_command`; even though I agree on the separation of concerns in this case.

```yaml
arg_command: ${{ github.event_name == 'merge_group' && 'apply' || 'plan -lock=false' }}
```

**Reply by rdhar** [reply]
Totally agreed, and I'm sure you're not the only one -- I preferred the previous single `command_input` parameter too.

But, like you say, separating out the command arguments into their own parameters is less ambiguous, even if a bit more verbose.

Thanks for opening this discussion!
